core/windows_and_tabs/window_snap.py

```diff
@@ -272,14 +276,38 @@ def __init__(self, left, top, right, bottom):
     "fullscreen": RelativeScreenPos(0, 0, 1, 1),
 }
 
+_split_positions = {
+    "split": [
+        _snap_positions["left third"],
+        _snap_positions["center third"],
+        _snap_positions["right third"],
+    ],
```

Comment:
```suggestion    
    "split": [
    [
        _snap_positions["left"],
        _snap_positions["right"],
    ],
    [
        _snap_positions["left third"],
        _snap_positions["center third"],
        _snap_positions["right third"],
    ]],
```

Let's make the values a list of lists, with different numbers of inputs. That way the data structure can support snapping layouts with 2/3/4/etc windows ergonomically. The action can then look at the number of windows that were passed, and find the layout that matches that length.

As an added bonus, if there isn't one, it can show a useful error explaining this ("layout 'clock' requires 3, 4, or 5 windows, but 2 were given")

The top level will still be string keys for simplicity (the spoken form) so we don't have to duplicate the list for each n-combination.

For the value, instead of a list of lists, we could use a dictionary with numeric keys (indicating the number of applications supported). But that's redundant given that the length of the value itself already has that information; it's less work for the user to just use list of lists. Make the code do the hard work of figuring out which one matches.

Reply:
I had a thought that I've talked myself out of: we could simplify this a bit further. Instead of needing to index into `_snap_positions` over and over, you can just use the string keys (`["left
 third", "center third", "right third"]`). The action implementation already has access to `_snap_positions` so it could do a simple array map to get those values. This would make the definitions shorter and less duplicative.

The only downside from my perspective is that in your current version you can use an anonymous `RelativeScreenPos` here without having to put it in `_snap_positions`, i.e. if there was a layout that only made sense as part of arrangement, not individual snapping. Requiring things be in `_snap_positions` means it would have to have a spoken form, which I think is a worse requirement.

You could do something where the values could be either string keys (in which case they are assumed to be `_snap_positions` keys and the action would look them up, or `RelativeScreenPos` objects (in which case the action would use them as is), but .... may not be worth the magic.

So the current way is fine, just posting this comment in case the line of thought is interesting.